### DISCUSSION TITLE
Link doesn't allow on production env

### DISCUSSION THREAD

**Discussion by Neonin** [amannn/next-intl#1095] [Q&A]
### Description

Need help, unfortunately I couldn't find a solution. Because it's very strange. 
Rekated on https://github.com/amannn/next-intl/issues/676

On my local machine, the links work as the documentation says, but when I deploy to the env vercel, I see nothing. Maybe I'm doing something wrong

My conifg
```typescript
const intlMiddleware = createMiddleware({
  locales,
  defaultLocale,
  localePrefix: 'as-needed',
})

export const config = {
  matcher: ['/((?!api|_vercel|_next/static|_next/image|favicon.ico|images|svg|.*\\..*).*)'],
}

``` 

![CleanShot 2024-05-28 at 00 16 29](https://github.com/amannn/next-intl/assets/16340041/92b7343a-6e13-4095-a03c-355449843502)
![CleanShot 2024-05-28 at 00 13 55](https://github.com/amannn/next-intl/assets/16340041/d73b8a64-4de6-4185-bb5d-52f008a55b64)

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

https://stage.klr.com.ua/en

### Reproduction description

Steps to reproduce:

1. Open reproduction
2. View head parameters (empty)

### Expected behaviour

Set the alternate link for an alternate language in response header

**Comment by amannn** [comment]
It seems like Vercel is stripping the `link` header here. I've noticed that Vercel seems to do this for cache hits, but haven't seen it on cache misses so far.

I'll move this to a discussion since it's not a bug of `next-intl`, but rather a hosting question, but I'd be interested in case you learn more about this. I'm unsure for example if Vercel would send different response headers if they're crawled by a search engine. If you want to be sure that your alternate pages are discovered, you can submit a [sitemap](https://next-intl-docs.vercel.app/docs/environments/metadata-route-handlers#sitemap) with alternate entries that create links between page versions.

**Reply by Neonin** [reply]
My SEO specialists still put me in the head to add links for canonical and alternate pages. 
Here is my workaround how I did it. But it's doesn't work for SSG😢

```typescript
import { headers } from 'next/headers'

import 'server-only'

import { defaultLocale, host, locales } from '../../../shared/configs'
import { AppLocales } from '../../../shared/constants/app-locales'

/**
 * Generates the canonical URL and languages based on the selected locale.
 * Temporary solution to generate canonical and languages
 * This function should be called on the server side
 *
 * @param selectedLocale The selected locale in app.
 * @returns An object containing the canonical URL and languages.
 */
export function generateCanonicalAndLanguages(selectedLocale: AppLocales) {
  const headersList = headers()
  // The `./middleware.ts` sets the x-current-url header in the server
  const pathname = headersList.get('x-current-url') ?? ''

  // Remove the locale from the pathname if it's the default locale
  const pathnameWithoutLocale =
    selectedLocale === defaultLocale ? pathname : pathname.replace(/\/[a-z]{2}?/, '')

  const path = pathnameWithoutLocale.includes('?')
    ? pathnameWithoutLocale.substring(0, pathname.indexOf('?'))
    : pathnameWithoutLocale

  const currentLocale = selectedLocale === defaultLocale ? '' : `/${selectedLocale}`

  const canonical = `${host}${currentLocale}${path}`

  const languages = locales.reduce(
    (acc, locale) => {
      acc[locale] = `${host}${defaultLocale === locale ? '' : `/${locale}`}${path}`
      return acc
    },
    {} as Record<AppLocales, string>
  )

  return {
    canonical,
    languages,
  }
}

``` 

`middleware.ts`
```typescript
export function middleware(req: NextRequest) {
  const { pathname } = req.nextUrl

  const response = intlMiddleware(req)
  response.headers.set('x-current-url', pathname)

  return response
}
``` 

`app/[locale]/layout`
```typescript
export async function generateMetadata({
  params: { locale },
}: LocaleLayoutParams): Promise<Metadata> {
  const t = await getTranslations({ locale, namespace: '' })

  return {
    alternates: generateCanonicalAndLanguages(locale),
  }
}
```
